packages/core/src/plugins/default/markbind-plugin-plantuml.ts

```diff
@@ -16,9 +16,32 @@ import { PluginContext } from '../Plugin';
 import { NodeProcessorConfig } from '../../html/NodeProcessor';
 import { MbNode } from '../../utils/node';
 
+interface DiagramStatus {
+  isAlive: boolean;
+}
+
+const LockManager = require('../../utils/LockManager');
+
 const JAR_PATH = path.resolve(__dirname, 'plantuml.jar');
 
-const processedDiagrams = new Set();
+const processedDiagrams = new Map<string, DiagramStatus>();
+
+function clearDeadDiagrams(diagrams: Map<string, DiagramStatus>) {
+  Array.from(diagrams.entries())
+    .filter(([, value]) => !value.isAlive)
+    .forEach(([key]) => diagrams.delete(key));
+}
+
+function initDiagrams(diagrams: Map<string, DiagramStatus>) {
```

Comment:
Could you explain why we are initializing the diagrams and setting their `isAlive` to false?
- what's the definition of `isAlive` ? Is it equivalent to `isLatest`?

Reply:
I have renamed it to isStale, used to track whether the diagram has been used by the site or latest diagram .